backend/infrastructures/apps/load/views/views.py

```diff
@@ -0,0 +1,97 @@
+import logging
+from pathlib import Path
+from typing import cast
+
+import inject
+from pandera.typing.pandas import DataFrame
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+
+from modules.common.domain import exceptions as domain_exceptions
+from modules.data.domain import value_objects as data_value_objects
+from modules.extract.domain import commands as domain_extract_commands
+from modules.extract.services import commands as service_extract_commands
+from modules.load.domain import commands as domain_load_commands
+from modules.load.domain.ports import units_of_work
+from modules.load.services import commands as services_load_commands
+from modules.load.services.queries import ports as query_ports
+from modules.transform.domain import commands as domain_transform_commands
+from modules.transform.services import commands as services_transform_commands
+
+from .serializers import ExtractDataSerializer, OutputDataSerializer
+
+logger = logging.getLogger(__name__)
+
+
+class DataViewSet(
+    ViewSet,
+):
+    @inject.param(name="save_data_unit_of_work", cls="save_data_unit_of_work")
+    def create(
+        self,
+        request: Request,
+        save_data_unit_of_work: units_of_work.AbstractDataUnitOfWork,
+    ) -> Response:
+        logger.info("Extracting Dataset...")
+
+        serializer = ExtractDataSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(
+                data=serializer.errors,
+                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+        try:
+            input_data: data_value_objects.InputData = cast(
+                data_value_objects.InputData,
+                service_extract_commands.extract(
+                    domain_extract_commands.ExtractData(
+                        Path(serializer.validated_data["file_path"])
+                    )
+                ),
+            )
+            logger.info("Transforming Dataset...")
+            output_data: list[
+                data_value_objects.OutputData
+            ] = services_transform_commands.transform(
+                domain_transform_commands.TransformData(cast(DataFrame, input_data))
+            )
+            logger.info("Saving Dataset...")
+            services_load_commands.save(
+                save_data_unit_of_work, domain_load_commands.SaveData(output_data)
+            )
+        except FileNotFoundError:
+            return Response(
```

Comment:
A good practise is to have structured and common way of returning error message.

E.g. infrastuctures/apps/common/consts.py add a 
ERROR_DETAIL_KEY = "detail"
and use it like this
```python
            logger.warning("'%s' is invalid format as location ID.", pk)
            raise exceptions.NotFound(
                {common_consts.ERROR_DETAIL_KEY: "Not found."}
            ) from err

        try:
```

Reply:
I'll do it tomorrow in next commits